### PR TITLE
Fix: Battle Bus With Passengers Freezes When Ordered To Attack Out Of Range Airborne Targets

### PR DESCRIPTION
**Pull request by commy2**
ZH 1.04

- The dummy weapon that allows the Battle Busses passangers to engage airborne targets also makes the Battle Bus freeze long before the Bus is in range when targeting airborne units.

After patch:

- When clicking on an enemy airborne target (e.g. Chinook), the Battle Bus will move into range, so that the troops inside can actually fire.

Note: `BattleBusDummyWeapon` is only used against airborne targets, despite the name being unclear about that.

### CODE REVIEW

File: Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini [hunk 1]
```diff
@@ -8467,7 +8467,7 @@ End
 Weapon BattleBusDummyWeapon
   PrimaryDamage = 0.0001
   PrimaryDamageRadius = 0.0
-  AttackRange = 320.0
+  AttackRange = 150.0 ; Patch104p @bugfix commy2 10/09/2021 Fix Bus freezing when ordered to attack out of range targets.
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini]:
```suggestion
  AttackRange = 150.0 ; Patch104p @bugfix commy2 10/09/2021 Fix Bus freezing when ordered to attack out of range airborne targets.
```

File: Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini [hunk 2]
```diff
@@ -8467,7 +8467,7 @@ End
 Weapon BattleBusDummyWeapon
   PrimaryDamage = 0.0001
   PrimaryDamageRadius = 0.0
-  AttackRange = 320.0
+  AttackRange = 150.0 ; Patch104p @bugfix commy2 10/09/2021 Fix Bus freezing when ordered to attack out of range airborne targets.
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini]: Same as the other change. How did this 150 value came to be?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini]: See: https://github.com/xezon/GeneralsGamePatch/pull/270#discussion_r706317490

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini]: See: https://github.com/xezon/GeneralsGamePatch/pull/270#discussion_r706321588

**Review by xezon — APPROVED** [review]

File: Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini (hunk 2 above)
Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini]:
```suggestion
  AttackRange = 150.0 ; Patch104p @bugfix commy2 10/09/2021 Fix Bus freezing when ordered to attack out of range airborne targets. 150.0 was determined to be the highest range where missile infantry passengers reliably engaged the target.
```

**Pull request merged**